### ISSUE TITLE
Writing (Saving) a terminal mode window to an existing file should fail and warn the user

### ISSUE THREAD

**Issue by peteyg** [neovim/neovim#13549]
- `nvim --version`: 0.4.4
- `vim -u DEFAULTS` (version: ) behaves differently? No (which is why I made this a feature request and not a bug)
- Operating system/version: Windows 10 version 1809
- Terminal name/version: both cmd and powershell
- `$TERM`: I don't think this exists on Windows

### Steps to reproduce using `nvim -u NORC`
Precondition: have an existing file (e.g. foo.txt) 
```
nvim -u NORC
:term
:w foo.txt
```

### Actual behaviour
Neovim happily writes the contents of the terminal buffer over foo.txt

### Expected behaviour
Neovim should abort the write and warn the user as it does when trying to overwrite an existing file from non-terminal buffers.

I say "should" because without this feature I typoed my way to losing a day's work and this addition would prevent others from suffering a similar fate. It also seems like this requested behavior would be consistent with how this situation is typically handled in other buffers.

**Comment by glacambre** [comment]
I like being able to write a terminal buffer's content to a file. How about adding the following to your init.vim instead?
```vim
autocmd TermOpen * autocmd BufWriteCmd <buffer> echoerr "Can't write terminal buffer"
```

**Comment by peteyg** [comment]
Just to be clear, I'm not suggesting to remove the ability entirely. I'm suggesting that it work like it does in other buffers where you're prompted to use `w!` instead.

I agree that writing buffers is a useful feature. I don't all think we should take that away.

**Comment by zeertzjq** [comment]
Fixed by #19375
